src/server/nyplApiClient/index.ts

```diff
@@ -0,0 +1,107 @@
+import NyplApiClient from "@nypl/nypl-data-api-client"
+import aws from "aws-sdk"
+
+interface KMSCache {
+  clients: string[]
+  clientId: string
+  clientSecret: string
+}
+
+const config: any = {
+  api: {
+    platform: {
+      development:
+        process.env.PLATFORM_API_BASE_URL ||
+        "https://qa-platform.nypl.org/api/v0.1",
+      production:
+        process.env.PLATFORM_API_BASE_URL ||
+        "https://platform.nypl.org/api/v0.1",
+    },
+    // The 'discovery' base URL should use DISCOVERY_API_BASE_URL if set,
+    // falling back on PLATFORM_API_BASE_URL if set,
+    // and finally falling back on a sensible default.
+    discovery: {
+      development:
+        process.env.DISCOVERY_API_BASE_URL ||
+        process.env.PLATFORM_API_BASE_URL ||
+        "https://qa-platform.nypl.org/api/v0.1",
+      production:
+        process.env.DISCOVERY_API_BASE_URL ||
+        process.env.PLATFORM_API_BASE_URL ||
+        "https://platform.nypl.org/api/v0.1",
+    },
+    drbb: {
+      development:
+        process.env.DRB_API_BASE_URL || "http://drb-api-qa.nypl.org/search/",
+      production:
+        process.env.DRB_API_BASE_URL ||
+        "https://digital-research-books-api.nypl.org/v3/sfr/search",
+    },
+  },
+  tokenUrl: "https://isso.nypl.org/",
+}
+const appEnvironment = process.env.APP_ENV || "production"
+
+const kms: aws.KMS = new aws.KMS({
+  region: "us-east-1",
+})
+
+const decryptKMS = async (key: string) => {
+  console.log("Decrypt func INPUT: " + key)
+  const params = {
+    CiphertextBlob: Buffer.from(key, "base64"),
+  }
+  const { Plaintext } = await kms.decrypt(params).promise()
+
+  return await new Promise((resolve, reject) => {
+    kms.decrypt(params, (err: Error) => {
+      if (err) {
+        console.log({ err })
+        reject(err)
+      } else {
+        // eslint-disable-next-line @typescript-eslint/no-base-to-string
+        resolve(Plaintext.toString())
+      }
+    })
+  })
+}
+
+const clientId = process.env.PLATFORM_API_CLIENT_ID
+const clientSecret = process.env.PLATFORM_API_CLIENT_SECRET
+
+const keys = [clientId, clientSecret]
+const CACHE: KMSCache = { clients: [], clientSecret, clientId }
```

Comment:
Super nitpick! (so safe to ignore)

I like placing const variables together and functions afterwards (since they tend to depend on variables defined before). In this case, these can be moved up near line 43 since that's another instance of defining a variable based on the env var.

Reply:
Good call! done.